src/tools/miri/src/shims/time.rs

```diff
@@ -362,6 +362,71 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         interp_ok(Scalar::from_i32(0))
     }
 
+    fn clock_nanosleep(
+        &mut self,
+        clock_id: &OpTy<'tcx>,
+        flags: &OpTy<'tcx>,
+        req_op: &OpTy<'tcx>,
+        _rem: &OpTy<'tcx>, // Signal handlers are not supported, so rem will never be written to.
+    ) -> InterpResult<'tcx, Scalar> {
+        let this = self.eval_context_mut();
+
+        let clock_id: libc::clockid_t = this.read_scalar(clock_id)?.to_i32()?;
```

Comment:
The first thing this function should do is `read_scalar` for all arguments (including `rem`). Only then are these arguments actually parsed an interpreted. That makes it easy to see all their types at a glance. See the [docs here](https://github.com/rust-lang/miri/blob/3814ea30c6b7e277b4b85432afa0b9d7bfad7b79/src/shims/foreign_items.rs#L261-L281) for some more detail.

(Seems like `nanosleep` also got this wrong, damn. That's what we get for relying on reviewers instead of tooling for enforcing a common coding style. ;)